src/public/assets/residential.js

```diff
@@ -0,0 +1,101 @@
+const agentlist = document.getElementById("agentlist");
+const lnamehead = document.getElementById("lnamehead");
+const fnamehead = document.getElementById("fnamehead");
+const feehead = document.getElementById("feehead");
+const ratinghead = document.getElementById("ratinghead");
+const regionDropdown = document.getElementById("regionDropdown")
+
+// Currency formatting
+const formatCurrency = new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: 'USD',
+    minimumFractionDigits: 2
+});
+
+// Function to create table rows
+const maketable = (data) => {
+    let agents = '';
+    data.forEach((e) => {
+        const row = agentlist.insertRow();
+        const first_name = e.first_name;
+        const last_name = e.last_name;
+        const fee = e.fee;
+        const rating = e.rating;
+        const dollarfee = formatCurrency.format(fee);
+
+
+        // Determine the class based on rating
+        let background_color = '';
+        if (rating == 100) {
+            background_color = "#1ceb0a"; /* bright green */
+        } else if (rating >= 90) {
+            background_color = "#42b1f1"; /* blue */        } 
+            else if (rating<90) {
+                background_color = "#A020F0"; /* purple  */        }
+
+
+        const rowdata = `
+        <tbody >
+            <tr style="background-color: ${background_color}">
+                <td><output readonly></output>${first_name}</td>
+                <td><output readonly></output>${last_name}</td>
+                <td><output readonly></output>${dollarfee}</td>
+                <td><output readonly></output>${rating}</td>
+            </tr>
+        </tbody>`;
+        agents += rowdata;
+
+        agentlist.innerHTML = agents;
+    });
+};
+
+// Fetch and filter agents data
+async function agentdata1() {
+    try {
+        const fetchreturn = await fetch('http://localhost:3004/topagents'); // Change to your backend URL
```

Comment:
You may want to switch this fetch to calling the basic all agents endpoint to ensure you do not have to account for any unexpected behavior on the topagents enpoint